derived_types.go

```diff
@@ -0,0 +1,261 @@
+package pgx
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+/*
+buildLoadDerivedTypesSQL generates the correct query for retrieving type information.
+
+	pgVersion: the major version of the PostgreSQL server
+	typeNames: the names of the types to load. If nil, load all types.
+*/
+func buildLoadDerivedTypesSQL(pgVersion int64, typeNames []string) string {
+	supportsMultirange := (pgVersion >= 14)
+	var typeNamesClause string
+
+	if typeNames == nil {
```

Comment:
Is this branch still valid? As far as I can see the only caller, LoadTypes, enforces len(typeNames) > 1.

Reply:
I would prefer this branch to be kept, as the alternative is to fail at runtime if a `nil` is passed in. It is an unexercised branch right now, and the compiler will optimise it away. 
If there were a way to make it a compile-time error, if a `nil` was passed in, it would be even better.

Reply:
I have modified this branch to run, but not match any records. This shouldn't usually happen, but at least avoids a runtime error.